tree/tree.h

```diff
@@ -0,0 +1,47 @@
+#ifndef _TREE_H__ 
+#define _TREE_H__
+
+#include "splitter.h"
+
+#include <vector>
+
+namespace tree {
+
+/**
+ * Base class for all Decision Tree implementations.
+ */
+class Tree{
+
+  public:
+
+    /**
+     * \brief Standard destructor.
+     */
+    virtual ~Tree() { }
+
+    /**
+     * Builds the Decision Tree
+     * \param X              An object of Collection
+     * \param sample_weight  Weight of each sample in collection
+     * \param feature_weight Weight of each feature 
```

Comment:
Is the \brief required here?
Correct me if I'm wrong, but I was reading the doxygen documentation again and since this is not the default javadoc style, I'm not sure why we are using the \ and the \brief.
Since everyone firstly agreed that we should use the javadoc, just remembering that it should be something like this:
`/**
* Brief description ending with dot.
* More elaborate description and etc.
* @param`
*/